pkg/controllers/management/clusterstats/statsaggregator.go

```diff
@@ -145,7 +145,10 @@ func (s *StatsAggregator) updateVersion(cluster *v3.Cluster) bool {
 			// This has the tendency to timeout
 			version, err := userContext.K8sClient.Discovery().ServerVersion()
 			if err == nil {
-				if cluster.Status.Version != version {
+				isClusterVersionOk := cluster.Status.Version != nil
+				isNewVersionOk := version != nil
+				if isClusterVersionOk != isNewVersionOk ||
+					(isClusterVersionOk && isNewVersionOk && *cluster.Status.Version != *version) {
```

Comment:
nit: you don't have to check both values not being nil on 151. The first condition pass would qualify for that